src/coprocessor/codec/convert.rs

```diff
@@ -261,6 +289,92 @@ pub fn convert_float_to_uint(ctx: &mut EvalContext, fval: f64, tp: FieldTypeTp)
     Ok(val as u64)
 }
 
+/// Converts a byte arrays to an u64 in best effort.
+#[inline]
+pub fn convert_bytes_to_uint(ctx: &mut EvalContext, bytes: &[u8], tp: FieldTypeTp) -> Result<u64> {
+    let s = str::from_utf8(bytes)?.trim();
+    let vs = get_valid_int_prefix(ctx, s)?;
+    let val = vs.parse::<u64>();
+    match val {
+        Ok(val) => convert_uint_to_uint(ctx, val, tp),
+        Err(_) => {
+            ctx.handle_overflow(Error::overflow("BIGINT UNSIGNED", &vs))?;
+            let val = integer_unsigned_upper_bound(tp);
+            Ok(val)
+        }
+    }
+}
+
+macro_rules! decimal_as_u64 {
+    ($expr:expr, $ctx:ident, $tp:ident) => {{
+        let val = match $expr.as_u64() {
+            Res::Ok(val) => val,
+            Res::Overflow(val) => {
+                $ctx.handle_overflow(Error::overflow("DECIMAL", ""))?;
+                val
+            }
+            Res::Truncated(val) => {
+                $ctx.handle_truncate(true)?;
+                val
+            }
+        };
+        convert_uint_to_uint($ctx, val, $tp)
+    }};
+}
+
+/// Converts a `DateTime` to an u64 value
+#[inline]
+pub fn convert_datetime_to_uint(
+    ctx: &mut EvalContext,
+    dt: &DateTime,
+    tp: FieldTypeTp,
+) -> Result<u64> {
+    // TODO: avoid this clone after refactor the `Time`
+    let mut t = dt.clone();
+    t.round_frac(DEFAULT_FSP)?;
+    decimal_as_u64!(t.to_decimal()?, ctx, tp)
+}
+
+/// Converts a `Duration` to an u64 value
+#[inline]
+pub fn convert_duration_to_uint(
+    ctx: &mut EvalContext,
+    dur: Duration,
+    tp: FieldTypeTp,
+) -> Result<u64> {
+    let dur = dur.round_frac(DEFAULT_FSP)?;
+    decimal_as_u64!(Decimal::try_from(dur)?, ctx, tp)
+}
+
+/// Converts a `Decimal` to an u64 value
+#[inline]
+pub fn convert_decimal_to_uint(
```

Comment:
There is many same code between `convert_decimal_to_uint` and `convert_decimal_to_int`.
BTW, this pattern appear serval time in this file
```rust
        Res::Ok(d) => d,
        Res::Overflow(d) => {
            ctx.handle_overflow(Error::overflow("DECIMAL", ""))?;
            d
        }
        Res::Truncated(d) => {
            ctx.handle_truncate(true)?;
            d
        }
```